### ISSUE TITLE
Error 10027: workers.api.error.script_too_large

### ISSUE THREAD

**Issue by evolve2k** [bitgenics/linc#34]
I'm using linc for the first time and am getting some strange deployment errors that mean that my site is not yet deploying corrrectly via cloudflare workers.

This error is displaying on my sites overview page:
`{"code":10027,"message":"workers.api.error.script_too_large"}`

Detailed version below:
```
Error releasing to Cloudflare Workers
There was an error releasing bundle b8bed0fc7 to Cloudflare Workers. We received the following error:

Fab deploy failed! Error uploading the script, got response: ❤️{"result":null,"success":false,"errors":[{"code":10027,"message":"workers.api.error.script_too_large"}],"messages":[]}❤️
To help diagnose the problem, first confi
```

I've checked the token permissions I have set and as the script is being generated by linc itself, Im thinking it's related to the 
script that linc itself creates being in this case **too large.**

**To Reproduce**
Any commit I push to my repo will generate an error currently.

**Expected behavior**
Pushing new commits, runs linc and undertakes a correct deploy.

**Screenshots**

![Overview___brave](https://user-images.githubusercontent.com/114459/88245992-402e4d80-ccd8-11ea-8a96-a094f3f4c58b.png)

**Comment by evanderkoogh** [comment]
Hey @evolve2k, unfortunately Cloudflare Workers has a 1Mb limit on script sizes: https://developers.cloudflare.com/workers/about/limits/#script-size

You exceeding that limit usually has 2 causes, you either have a lot of server-side code that needs to be included (including stubbing out some NodeJS libraries), or you are including a lot of static HTML files.

If you have a lot of HTML files, we currently inline all of them because they are turned into templates on compile, so we can inject tags etc. @geelen has an issue on the FAB roadmap that would fix that issue, so if that is you, please let us know so we can prioritise that.

If that isn't the case, you probably have a lot of server-side code (in something like NextJS or something similar) and you may be able to switch out libraries. Switching out Axios for native fetch for example would remove a chunk of dependencies.

Your last resort is to host your code somewhere else, such as maybe AWS Lambda@Edge, their limits very different.

And if you want, @geelen or I would be more than happy to take a look at your setup if you invite us to the repo, or just send us a FAB.

**Comment by evolve2k** [comment]
Thanks @evanderkoogh.

### 1. Here's access to my repo
@geelen hope you're going well mate, I'd love you to have a look at my repo.
I've now given @evanderkoogh and yourself access to the repo.
It's a basic Nuxt Vue App that is set to generate a static site that I then bundle up into a fab.

### 2. I have another cloudflare script running to set Secure Headers

I currently have one script already running in cloudflare, basically just sets Security Headers as recommended by Scott Helme here: https://scotthelme.co.uk/security-headers-cloudflare-worker

However doesn't sound like I can't have two scripts running, just that they each need to be under 1mb.
Mentioning just incase.

**Comment by evanderkoogh** [comment]
Hey @evolve2k, yeah it is exactly what I thought. We currently inline all your HTML files into the server file that gets deployed to Cloudflare, so it will be too big indeed.
I think @geelen is about to work on externalising those HTML files, so we can support larger sites on things like Cloudflare Workers.

And the security headers situation is *exactly* why I am so freaking excited about FABs. It allows you to do what we have taken to call Static+. A mostly static site, but with little dynamic things sprinkled in, such as headers, working with cookies, proxying backend requests and the list is just endless.

I am chatting with @geelen tonight and we will be discussing what we will be working on this week, so we will get back with you tomorrow.

**Comment by evolve2k** [comment]
Thanks @evanderkoogh @geelen you folks are awesome!

**Comment by evanderkoogh** [comment]
Hey @evolve2k, @geelen got around to working on this yesterday. See https://github.com/fab-spec/fab/pull/214

It is released as `v1.0.0-rc6`. Would love for you to give it a go and let us know if that fixed the problem.

**Comment by evolve2k** [comment]
Hi Erwin, sorry for the delay.

I've gone and tested it and can't get it working.
Initially I thought it was because maybe I'd made other code changes and
I've spent time going through and attempting to debug this myself and in
the process have made a
stack of test branches and minor code changes towards attempting to fix the
build errors.

The code is breaking with this error:

   - [fatal] Nuxt build error
   - ERROR in ./.nuxt/App.js
   - Module not found: Error: Can't resolve 'sass-loader' in
   '/opt/buildhome/build/src'
   - @ ./.nuxt/App.js 27:0-69
   - @ ./.nuxt/index.js
   - @ ./.nuxt/client.js
   - @ multi ./.nuxt/client.js


But I've done everything I could think of to address a sass-loader issue
and further it's all working fine on my local machine -- Works on my
machine ;)

I've finally stopped and created a new project from scratch
called brave-sliced and just got it setup and deployed to linc with fab.
That's also generating errors, desipte fab working fine locally so Im
now inclined to think its on your end.

I just didnt want to tell you I'd tested it if the issue was on my end.

Please have a look again, and apologies if I've generated errors on my end
and that's what's breaking it now.

Richie

**Comment by evanderkoogh** [comment]
Hey @evolve2k, the problem is always at our end. Either there is a bug, or we haven't explained well enough how to do it properly :)
@geelen will take a look at this as soon as possible and let you know which one of those two things it is.

**Comment by geelen** [comment]
Hey @evolve2k, I... _think_ this is going to be a missing library on our end, though I'm really confused as to how you're ending up with that message so I might be wrong. `sass-loader` is a common problem though since it has some C library dependencies (it's one of the few JS libraries that does).

Still, that error message gives me pause... I would have expected the `npm ci` step to fail...

I'm afraid I won't get to it today but Erwin will take a look at adding `libsass` to the build image tomorrow and see if that magically makes it work. Either that or it's something weird with our `devDependencies` again...

**Comment by evanderkoogh** [comment]
Hey @evolve2k, apologies for not getting back to you earlier. But I finally had some time to dig around properly and I realised that the problem is that you set `NODE_ENV` to `production` in your build configuration. We already do that automatically for every call, except for `npm ci` and related installation commands, but if you run those with `NODE_ENV=production` they do not install the `devDependecies`.

And unfortunately we let your build configuration override our hardcoded `NODE_ENV=development`. This is now fixed, I could successfully build your site. The Docker image is now uploading, which is going to take a bit on Australian interwebs, but should be there in ~10 min :)

Sorry for the inconvenience and thanks for reporting this!

**Comment by evolve2k** [comment]
Thanks for your efforts with this.

@evanderkoogh whatever you did with your latest changes definitley helped and a few more of my builds are passing now, so that's great.

My main master branch is still having issues though. 
$ npm run build:fab
$ npm ci 

both run without problems locally.

Per the linc log the build seems to run the whole way and then just fails in the last step.

```
...
[Compiler] Compiling your server.js:
[Compiler] Done in 0.55 seconds.
[Generator] Writing all files to .fab/build
   ... lists assets
[Generator] Done in 0.03 seconds.
[Generator] Zipping it up into a FAB:
[Generator] Created fab.zip (2.41 MB) in 0.15 seconds
Error creating FAB bundle from /opt/buildhome/build/src/fab.zip.
```

**Comment by evanderkoogh** [comment]
That is really weird. Especially that it works on non-master branches and not on master? Could you try rerunning one? We just upgraded the memory significantly on the build machines, because some people were having issues with memory allocation.

**Comment by evolve2k** [comment]
Oh I think it's not that it does or doesn't work on master/non-master branches, more that the collective code on my master branch is producing that output whereas branches from earlier that I have pushed out, built after your change when I attempted to rebuild them.

And I have no way of debugging why my latest master code is not working.. as it build fine locally.. and the error is unclear in linc as you can see above.

**Comment by evanderkoogh** [comment]
Hmm.. Some more investigation reveals it is not an error building, but deploying the FAB. I am on it :)

**Comment by evanderkoogh** [comment]
Ok @evolve2k, I figured out what the problem is. This could go straight into that Alanis song. The irony is that it fails on updating the database with the successful build information.. 

I know why it goes wrong, just have to figure out how to refactor a piece of code. Will be fixed tomorrow at the latest, lockdown permitting :)

**Comment by evanderkoogh** [comment]
Hey @evolve2k, I have fixed that particular issue, but now all your builds just hang on `nuxt generate` indefinitely on node 13 and higher, and fail with a `Date` error on node 12 and below.

This happens on Linc, in the Docker container locally and when I check out the local source code. Could you check what happens to you on the last master commit and if it works what version of Node you are running?

**Comment by evolve2k** [comment]
Hi @evanderkoogh Im not totally sure what's been happening here.

Prismic Nuxt recommends two different approaches to getting started.
Their main webpage; basically leads to [this tutorial on their homepage](https://user-guides.prismic.io/en/articles/2831943-nuxt-js-sample-multi-page-website-with-navigation) and [this example app repo](https://github.com/prismicio/nuxtjs-website)

But their new SliceMachine.dev site that leverages Nuxt and recommends their new cli

    npx create-nuxt-app my-slicemachine-project

ref: https://www.slicemachine.dev/documentation/nuxt/nuxt-quick-start

I've started a new repo and followed **npx create-nuxt-app** and have created a project that builds with linc.
Basically using their new cli tool to  create-nuxt-app seems to have improved everything and its solved the build issues I was facing, but probably you resolving the script to large issue was still needed.

If users use npx create-nuxt-app they'll be able to create a linc project that builds.

To test getting nuxt (and prismic) and linc working together, you can follow this tute:
https://www.slicemachine.dev/documentation/nuxt

_I have another question, but it's different, so i'll open a new ticket_

**I think you're pretty much good to add Nuxt as a platform that works with linc now.**
